tests/tt_metal/tt_metal/unit_tests_fast_dispatch_single_chip_multi_queue/command_queue/test_EnqueueQueueWaitForEvent.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <memory>
+
+#include "command_queue_fixture.hpp"
+#include "common/logger.hpp"
+#include "gtest/gtest.h"
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/test_utils/env_vars.hpp"
+#include "tt_metal/detail/tt_metal.hpp"
+
+using namespace tt::tt_metal;
+
+namespace local_test_functions {
+
+void FinishAllCqs(vector<std::reference_wrapper<CommandQueue>>& cqs) {
+    for (uint i = 0; i < cqs.size(); i++) {
+        Finish(cqs[i]);
+    }
+}
+
+}
+
+namespace basic_tests {
+
+// Simplest test to record Event per CQ and wait from host.
+TEST_F(MultiCommandQueueSingleDeviceFixture, TestEventsEventSynchronizeSanity) {
+    CommandQueue a(this->device_, 0);
+    CommandQueue b(this->device_, 1);
+    vector<std::reference_wrapper<CommandQueue>> cqs = {a, b};
+    std::unordered_map<uint, std::vector<Event>> sync_events;
+    size_t num_events = 10;
+
+    for (size_t j = 0; j < num_events; j++) {
+        for (uint i = 0; i < cqs.size(); i++) {
+            log_debug(tt::LogTest, "Recording and Host Syncing on event for CQ ID: {}", cqs[i].get().id());
+            auto &event = sync_events[i].emplace_back(Event());
+            EnqueueQueueRecordEvent(cqs[i], event);
+            EXPECT_EQ(event.cq_id, cqs[i].get().id());
+            EXPECT_EQ(event.event_id, j); // 1 cmd per CQ
+            EventSynchronize(event);
+        }
+    }
+
+    // Sync on some earlier events per CQ just to show it works.
+    for (uint i = 0; i < cqs.size(); i++) {
+        EventSynchronize(sync_events.at(i)[3]);
```

Comment:
3, 4 made me pause, both are <10, maybe use constexpr for them and assert they are less than num_events?

Reply:
Yeah, that's fair. Not worth the trouble to come up with some funky expression, I just changed it to loop over all events per CQ again, and sync on them in order per CQ (instead of bouncing between CQs), pushing in few mins after rebase/build tested locally.